modules/renderer/features.js

```diff
@@ -479,8 +484,9 @@ export function rendererFeatures(context) {
 
 
     features.isHiddenFeature = function(entity, resolver, geometry) {
-        if (!_hidden.length) return false;
         if (!entity.version) return false;
+        if (!features.featureFitsDateRange(entity)) return true;
+        if (!_hidden.length) return false;
         if (_forceVisible[entity.id]) return false;
```

Comment:
If you select a feature then enable a filter, the feature remains visible until you deselect the feature. This isn’t happening for date-filtered features because this check is coming after the call to `featureFitsDateRange()`.

![nodes](https://user-images.githubusercontent.com/1231218/194684677-ce1bda18-ae9d-4bae-8af0-d18edcfb6afa.png)